app/controllers/restaurants_controller.rb

```diff
@@ -3,6 +3,9 @@ class RestaurantsController < ApplicationController
   before_action :check_for_exisiting_restaurant, only: :new
 
   def index
+    if Restaurant.any?
+      gon.restaurants = Restaurant.all
+    end
```

Comment:
Solved by not running gon if no gon is called.
And only calling gon if there are restaurants in it. 
But perhaps I did something wrong in that commit?